stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,184 @@
+    ) -> tuple[AnyStr, int]: ...
+    def scanner(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        overlapped: bool = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Scanner[AnyStr]: ...
+
+class Match(Generic[AnyStr]):
+
+    re: Pattern[AnyStr]
+    string: AnyStr
+    pos: int
+    endpos: int
+    partial: bool
+    regs: tuple[tuple[int, int], ...]
+    fuzzy_counts: tuple[int, int, int]
+    fuzzy_changes: tuple[list[int], list[int], list[int]]
+    lastgroup: str | None
+    lastindex: int | None
+    @overload
+    def group(self, __group: int | str = ...) -> AnyStr | None: ...
+    @overload
+    def group(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[AnyStr | None, ...]: ...
+    @overload
+    def groups(self, default: None = ...) -> tuple[AnyStr | None, ...]: ...
+    @overload
+    def groups(self, default: _T) -> tuple[AnyStr | _T, ...]: ...
+    @overload
+    def groupdict(self, default: None = ...) -> dict[str, AnyStr | None]: ...
+    @overload
+    def groupdict(self, default: _T) -> dict[str, AnyStr | _T]: ...
+    @overload
+    def span(self, __group: int | str = ...) -> tuple[int, int]: ...
+    @overload
+    def span(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[tuple[int, int], ...]: ...
+    @overload
+    def spans(self, __group: int | str = ...) -> list[tuple[int, int]]: ...
+    @overload
+    def spans(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[list[tuple[int, int]], ...]: ...
+    @overload
+    def start(self, __group: int | str = ...) -> int: ...
+    @overload
+    def start(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[int, ...]: ...
+    @overload
+    def starts(self, __group: int | str = ...) -> list[int]: ...
+    @overload
+    def starts(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[list[int], ...]: ...
+    @overload
+    def end(self, __group: int | str = ...) -> int: ...
+    @overload
+    def end(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[int, ...]: ...
+    @overload
+    def ends(self, __group: int | str = ...) -> list[int]: ...
+    @overload
+    def ends(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[list[int], ...]: ...
+    def expand(self, template: AnyStr) -> AnyStr: ...
+    def expandf(self, format: AnyStr) -> AnyStr: ...
+    @overload
+    def captures(self, __group: int | str = ...) -> list[AnyStr]: ...
+    @overload
+    def captures(self, __group1: int | str, __group2: int | str, *groups: int | str) -> tuple[list[AnyStr], ...]: ...
+    def capturesdict(self) -> dict[str, list[AnyStr]]: ...
+    def detach_string(self) -> None: ...
+
+class Splitter(Generic[AnyStr]):
+
+    pattern: Pattern[AnyStr]
+    def __iter__(self) -> Iterator[AnyStr]: ...
```

Comment:
`_typeshed.Self` would be useful for this, as it returns the object unchanged:

```suggestion
    def __iter__(self: Self) -> Self: ...
```

You will need to add an import.

Reply:
updated this method and the `__iter__` method in `regex/_regex/Scanner` to use this 👍